src/thor/trace_route_action.cc

```diff
@@ -389,12 +388,32 @@ void thor_worker_t::build_route(
     add_path_edge(destination_location, dest_segment->edgeid, dest_segment->target, dest_match.lnglat,
                   dest_match.distance_from);
 
+    // build up the discontinuities so we can trim shape where we do uturns
+    edge_trimming.clear();
+    for (size_t i = 0; i < path.second.size(); ++i) {
+      const auto* prev_segment = i > 0 ? path.second[i - 1] : nullptr;
```

Comment:
Nit: For readability in the diff, these should be the actual type and not `auto`.